src/musicbrainz/web/coverartarchivelinkstask.cpp

```diff
@@ -0,0 +1,157 @@
+
+const QUrl kBaseUrl = QStringLiteral("https://coverartarchive.org/");
+
+const QString kRequestPath = QStringLiteral("/release/");
+
+network::JsonWebRequest lookupRequest() {
+    return network::JsonWebRequest{
+            network::HttpRequestMethod::Get,
+            kRequestPath,
+            QUrlQuery(),
+            QJsonDocument()};
+}
+
+QNetworkRequest createNetworkRequest(
+        const QUuid& releaseID) {
+    DEBUG_ASSERT(kBaseUrl.isValid());
+    DEBUG_ASSERT(!releaseID.isNull());
+    QUrl url = kBaseUrl;
+    url.setPath(kRequestPath + releaseID.toString(QUuid::WithoutBraces));
+    DEBUG_ASSERT(url.isValid());
+    QNetworkRequest networkRequest(url);
+    return networkRequest;
+}
+
+} // anonymous namespace
+
+CoverArtArchiveLinksTask::CoverArtArchiveLinksTask(
+        QNetworkAccessManager* pNetworkAccessManager,
+        const QUuid& albumReleaseId,
+        QObject* pParent)
+        : network::JsonWebTask(
+                  pNetworkAccessManager,
+                  kBaseUrl,
+                  lookupRequest(),
+                  pParent),
+          m_albumReleaseId(albumReleaseId) {
+}
+
+QNetworkReply* CoverArtArchiveLinksTask::sendNetworkRequest(
+        QNetworkAccessManager* pNetworkAccessManager,
+        network::HttpRequestMethod method,
+        const QUrl& url,
+        const QJsonDocument& content) {
+    DEBUG_ASSERT(pNetworkAccessManager);
+    Q_UNUSED(method);
+    DEBUG_ASSERT(method == network::HttpRequestMethod::Get);
+    pNetworkAccessManager->setRedirectPolicy(QNetworkRequest::NoLessSafeRedirectPolicy);
+
+    const QNetworkRequest networkRequest = createNetworkRequest(m_albumReleaseId);
+
+    VERIFY_OR_DEBUG_ASSERT(url.isValid()) {
+        kLogger.warning() << "Invalid URL" << url;
+        return nullptr;
+    }
+    if (kLogger.traceEnabled()) {
+        kLogger.trace()
+                << "Get"
+                << url;
+    }
+    return pNetworkAccessManager->get(networkRequest);
+}
+
+void CoverArtArchiveLinksTask::onFinished(
+        const network::JsonWebResponse& response) {
+    if (!response.isStatusCodeSuccess()) {
+        kLogger.warning()
+                << "Request failed with HTTP status code"
+                << response.statusCode();
+        emitFailed(response);
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(response.statusCode() == network::kHttpStatusCodeOk) {
+        kLogger.warning()
+                << "Unexpected HTTP status code"
+                << response.statusCode();
+        emitFailed(response);
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(response.content().isObject()) {
+        kLogger.warning()
+                << "Invalid JSON content"
+                << response.content();
+        emitFailed(response);
+        return;
+    }
+    const auto jsonObject = response.content().object();
+
+    if (jsonObject.isEmpty()) {
+        kLogger.warning()
+                << "Empty Json";
+        emitFailed(response);
+        return;
+    }
+
```

Comment:
I will simply add a link to a reference (A release with the most cover arts) instead of the whole JSON, that will look a bit better in shape, what do you think?

Reply:
The link to a real Json is a good idea.
Please add also:
https://wiki.musicbrainz.org/Cover_Art_Archive/API